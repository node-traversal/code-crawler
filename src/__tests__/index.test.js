const path = require('path');
const scanner = require('../index');
const reactPlugin = require('../parser-plugin-react');

const basePath = process.cwd();
console.log('basePath', basePath);
const examplesPath = path.join(basePath, 'examples/simple');

describe('The scanner should', () => {
    it('parse simple code', (cb) => {
        scanner(examplesPath, [reactPlugin], (results) => {
            expect(results).toMatchSnapshot();
            cb();
        });
    });
});
