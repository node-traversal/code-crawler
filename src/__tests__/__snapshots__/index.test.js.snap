// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The scanner should parse simple code 1`] = `
Object {
  "undefined": Object {
    "deps": Array [
      Object {
        "name": "react",
        "var": "React",
      },
      Object {
        "name": "prop-types",
        "var": "PropTypes",
      },
    ],
    "props": Array [
      Object {
        "name": "firstName",
        "type": "string",
      },
    ],
  },
}
`;
